client/scripts/views/components/admin_panel.ts

```diff
@@ -0,0 +1,634 @@
+import m, { Vnode } from 'mithril';
+import $ from 'jquery';
+import { OffchainThread, OffchainTag, CommunityInfo, RolePermission, ChainInfo, ChainNetwork, RoleInfo } from 'models';
+import { Button, Classes, Dialog, Icon, Icons, Tag, TagInput, ListItem, Table, Input, List, TextArea, Switch, Tabs, TabItem, RadioGroup, Form, FormGroup, FormLabel } from 'construct-ui';
+import app from 'state';
+import { sortAdminsAndModsFirst } from 'views/pages/discussions/roles';
+import User from './widgets/user';
+import 'components/admin_panel.scss';
+
+const RoleRow: m.Component<{ roledata?, onRoleUpdate?: Function }> = {
+  view: (vnode) => {
+    return (vnode.attrs.roledata?.length > 0) &&
+      m('RoleData', [
+          vnode.attrs.roledata?.map((role) => {
+            const chainOrCommObj = (app.activeChainId())
+              ? { chain: app.activeChainId() }
+              : { community: app.activeCommunityId() };
+            return m('.role-item', { style: 'display: inline-block; padding-right: 6px;' }, [
+              m(User, {
+                user: [role.Address.address, role.Address.chain],
+                linkify: true,
+                tooltip: true,
+              }),
+              m(Icon, {
+                name: Icons.X,
+                size: 'xs',
+                style: 'padding-left: 2px;',
+                onclick: () => {
+                  $.post(`${app.serverUrl()}/upgradeMember`, {
+                    ...chainOrCommObj,
+                    new_role: 'member',
+                    address: role.Address.address,
+                    jwt: app.login.jwt,
+                  }).then((res) => {
+                    if (res.status !== 'Success') {
+                      throw new Error(`got unsuccessful status: ${res.status}`);
+                    }
+                    vnode.attrs.onRoleUpdate(role, res.result);
+                  }).catch((e) => console.error('Failed To demote admin'));
+                },
+              }),
+            ]);
+          })
+      ]);
+  }
+};
+
+interface ICommunityMetadataState {
+    name: string;
+    description: string;
+    url: string;
+    privacyToggled: boolean;
+    invitesToggled: boolean;
+}
+
+interface IChainCommunityAttrs {
+  community?: CommunityInfo;
+  chain?: ChainInfo;
+  onChangeHandler: Function;
+  onRoleUpdate: Function;
+  admins;
+  mods;
+}
+
+interface ITableRowAttrs {
+  title: string;
+  defaultValue: string;
+  disabled?: boolean;
+  onChangeHandler: Function;
+}
+
+const TableRow: m.Component<ITableRowAttrs> = {
```

Comment:
Components should be named **and classed** with a name that is as specific as possible for the scope they are in (in this case, this table is for the community management panel, so perhaps that). Otherwise the styling will conflict with any other classes named TableRow.

This should also apply to any other components in this file, e.g. RoleRow -> CommunityManagementRoleRow